lib/wallet/src/Cardano/Wallet.hs

```diff
@@ -1255,6 +1257,24 @@ shelleyOnlyMkSelfWithdrawal netLayer txWitnessTag era db =
     notShelleyWallet = throwIO
         $ ExceptionReadRewardAccount ErrReadRewardAccountNotAShelleyWallet
 
+mkSelfWithdrawalShared
+    :: forall n block
+     . NetworkLayer IO block
+    -> TxWitnessTag
+    -> AnyCardanoEra
+    -> DBLayer IO (SharedState n SharedKey)
+    -> IO Withdrawal
+mkSelfWithdrawalShared netLayer txWitnessTag era db = do
+    rewardAccountM <- readSharedRewardAccount db
+    when (isNothing rewardAccountM) $
+        throwIO $ ExceptionReadRewardAccount ErrReadRewardAccountMissing
+    let (rewardAccount, derivationPath) = fromJust rewardAccountM
+    balance <- getCachedRewardAccountBalance netLayer rewardAccount
+    pp <- currentProtocolParameters netLayer
+    return $ case checkRewardIsWorthTxCost txWitnessTag pp era balance of
```

Comment:
I suppose you could

## A 

Pass in the `delegationTemplate` `checkRewardIsWorthTxCost`.  We'd then also need to update the logic here https://github.com/input-output-hk/cardano-wallet/blob/ca7eba61eb11b576d6e20a2237955c064ecd2b7a/lib/wallet/src/Cardano/Wallet/Shelley/Transaction.hs#L1774 which would be antithetical to having `estimateTxSize` know the bare minimum to support coin-selection. 

# B 

Define

```haskell
-- | Upper estimation of the number of bytes needed for a withdrawal.
txRewardWithdrawalSize
    :: WalletState s
    -> TxSize
```

which would be more in line with https://input-output.atlassian.net/browse/ADP-3016

# C

Do nothing, or raise hard-code limit of `2 * costOfWithdrawal`.

This would likely work fine for most users. Users with large delegation templates could lose money from withdrawals which are too small to be worth it. Although I believe it would be on the order of 0.01 ada, and at most once per epoch... So probably completely fine...

Reply:
opted for A

Reply:
Unless you also update the `defaultTransactionCtx` in `checkRewardIsWorthTxCost`, your changes have no effect

https://github.com/input-output-hk/cardano-wallet/blob/f1757d4a398bc200930b316d64f82ec4e4a44a42/lib/wallet/src/Cardano/Wallet.hs#L1287-L1288

Reply:
done